include/posix/time.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef __POSIX_TIME_H__
+#define __POSIX_TIME_H__
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifdef CONFIG_NEWLIB_LIBC
+#include_next <time.h>
+#else
+struct timespec {
+	s32_t tv_sec;
+	s32_t tv_nsec;
+};
+#endif /* CONFIG_NEWLIB_LIBC */
+#include <kernel.h>
+#include <errno.h>
+#include "sys/types.h"
+
+#define CLOCK_MONOTONIC 1
+
+int clock_gettime(clockid_t clock_id, struct timespec *ts);
+
+/**
+ * @brief POSIX threading compatibility API
+ *
+ * See IEEE 1003.1
+ */
+static inline int clock_settime(clockid_t clock_id, const struct timespec *ts)
+{
+	errno = (clock_id != CLOCK_MONOTONIC) ? EINVAL : ENOSYS;
```

Comment:
The error here should always be `ENOSYS`, since it's just a stub. Returning `EINVAL` does not help the user figure out what's wrong.